gkeys/gkeys/email_reminders.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import print_function
+
+import os
+import re
+import smtplib
+import sys
+
+from string import Template
+
+from email.mime.text import MIMEText
+from snakeoil.demandload import demandload
+
+if sys.version_info[0] >= 3:
+    py_input = input
+    _unicode = str
+else:
+    py_input = raw_input
+    _unicode = unicode
+
+demandload(
+    "gkeys.base:Args",
+    "json:load",
+)
+
+class Emailer(object):
+    '''Send an email reminder about the status of the user's GPG key'''
+
+    def __init__(self, results, gkey):
+        uid = ''
+        if gkey.uid:
+            uid = gkey.uid[0]
+            match = re.findall(r'[\w\.-]+@[\w\.-]+', uid)
+            uid = ''
+            if match:
+                uid = match[0]
+        self.uid = uid
+        self.sub = []
+        for key in results:
+            print(results[key])
+            for specs in results[key]:
+                if specs.key == 'PUB':
+                    self.pub = specs
+                else:
+                    self.sub.append(specs)
+
+    def _send_email(self, message):
+        mail = smtplib.SMTP('smtp.gmail.com', 587)
+        mail.ehlo()
+        mail.starttls()
+        mail.login('aperivol@gmail.com', 'PASSWORD')
+        subject = "Expiring Key"
+        email_to = "justthisthing@gmail.com" #uid
+        email_from = "aperivol@gmail.com"
+        msg = MIMEText(message, 'plain')
+        msg['Subject'] = subject
+        msg['From'] = email_from
+        mail.sendmail('aperivol@gmail.com', 'justthisthing@gmail.com', msg.as_string())
+        mail.quit()
+
+    def _generate_template(self, template_path, keyprints, specprint):
+        with open(template_path, 'r') as file_contents:
+            content = file_contents.read()
+        message_template = Template(content)
+        subkey_msg = ''
+        message = message_template.substitute(key_print=keyprints, spec_print=specprint)
+        return message
+
+    def _is_expiring(self):
+        is_expiring = False
+        if self.pub.days < 180 and self.pub.days > -180:
+            is_expiring = True
+        else:
+            for subkey in self.sub:
+                if subkey.days < 180 and subkey.days > -180:
+                    is_expiring = True
+                    break
+        return is_expiring
+
+    def email_reminder(self, keyprints, specprint):
+        is_expiring = self._is_expiring()
+        if self.uid and is_expiring:
+            template_path = '/usr/share/gkeys/templates/email_template'
+            if os.path.exists('/home/aeroniero/.config/gkeys/templates/email_template'):
+                template_path = '/home/aeroniero/.config/gkeys/templates/email_template'
+            elif os.path.exists('/var/lib/gentoo/gkeys/templates/email_template'):
+                template_path = '/var/lib/gentoo/gkeys/templates/email_template'
+            message = self._generate_template(template_path, keyprints, specprint)
+            self._send_email(message)
```

Comment:
Again with hard-coded paths.  All these paths should be configured...

[templates]
template_dir = foo
expiry = %(template_dir)s/bar

the /var/... templates is not needed unless we are also having dynamic templates that are downloaded.

so the /usr/share/... and user  templates are enough.

Also for the user templates, use the config values for the location. look at how it determines the user config location during the startup of the script.